lib/src/flutterflow_api_client.dart

```diff
@@ -265,3 +279,72 @@ Future _runFix({
     stderr.write('Error running "dart fix": $e\n');
   }
 }
+
+Future firebaseDeploy({
+  required String token,
+  required String projectId,
+  required String destinationPath,
+  String endpoint = kDefaultEndpoint,
+}) async {
+  final endpointUrl = Uri.parse(endpoint);
+  final body = jsonEncode({
+    'project': {
+      'path': 'projects/$projectId',
+    },
+    'token': token,
+  });
+  final result = await _callEndpoint(
+    client: http.Client(),
+    token: token,
+    url: Uri.https(
+        endpointUrl.host, '${endpointUrl.path}/exportFirebaseDeployCode'),
+    body: body,
+  );
+
+  // Download actual code
+  final projectZipBytes = base64Decode(result['firebase_zip']);
+  final firebaseProjectId = result['firebase_project_id'];
+  final projectFolder = ZipDecoder().decodeBytes(projectZipBytes);
+  extractArchiveToCurrentDirectory(projectFolder, destinationPath);
+  final firebaseDir = '$destinationPath/firebase';
+
+  // Install required modules for deployment.
+  await Process.run(
+    'npm',
+    ['install'],
+    workingDirectory: '$firebaseDir/functions',
+    runInShell: true,
+    stdoutEncoding: utf8,
+    stderrEncoding: utf8,
+  );
+  final directoriesResult = await Process.run(
+    'ls',
+    [],
+    workingDirectory: firebaseDir,
+    runInShell: true,
+    stdoutEncoding: utf8,
+    stderrEncoding: utf8,
+  );
+
+  // This directory only exists if there were custom cloud functions.
+  if (directoriesResult.stdout.contains('custom_cloud_functions')) {
+    await Process.run(
+      'npm',
+      ['install'],
+      workingDirectory: '$firebaseDir/custom_cloud_functions',
+      runInShell: true,
+      stdoutEncoding: utf8,
+      stderrEncoding: utf8,
+    );
+  }
+
+  final deployProcess = await Process.start(
+    'firebase',
+    ['deploy', '--project', firebaseProjectId],
```

Comment:
One thing that worries me is Firestore Rules. Imagine you host many FlutterFlow projects in one Firebase project, or you have some existing collections in your Firebase project, not managed by FlutterFlow. Then when you deploy to Firebase this way, it'll overwrite all existing Firestore rules with rules for this FlutterFlow project. Right?

When you deploy via FlutterFlow frontend it's more intelligent: it won't overwrite rules for collections it doesn't recognize. Perhaps `exportFirebaseDeployCode` should do the same.

Reply:
Added `--append-rules` flag